core/src/main/scala/scalaz/syntax/ApplySyntax.scala

```diff
@@ -114,5 +174,65 @@ trait ApplySyntax[F[_]] extends FunctorSyntax[F] {
+                                             f: (A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T) => U): F[U] =
+    F.apply17(fa, fb, fc, fd, fe, fi, fj, fk, fl, fm, fn, fo, fp, fq, fr, fs, ft)(f)
+
+  def ^^^^^^^^^^^^^^^^^[A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T,U,V](fa: => F[A], fb: => F[B], fc: => F[C], fd: => F[D], fe: => F[E], fi: => F[I], fj: => F[J], fk: => F[K], fl: => F[L], fm: => F[M], fn: => F[N], fo: => F[O], fp: => F[P], fq: => F[Q], fr: => F[R], fs: => F[S], ft: => F[T], fu: => F[U])(
+                                               f: (A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T,U) => V): F[V] =
+    F.apply18(fa, fb, fc, fd, fe, fi, fj, fk, fl, fm, fn, fo, fp, fq, fr, fs, ft, fu)(f)
+
+  def ^^^^^^^^^^^^^^^^^^[A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W](fa: => F[A], fb: => F[B], fc: => F[C], fd: => F[D], fe: => F[E], fi: => F[I], fj: => F[J], fk: => F[K], fl: => F[L], fm: => F[M], fn: => F[N], fo: => F[O], fp: => F[P], fq: => F[Q], fr: => F[R], fs: => F[S], ft: => F[T], fu: => F[U], fv: => F[V])(
+                                                 f: (A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T,U,V) => W): F[W] =
+    F.apply19(fa, fb, fc, fd, fe, fi, fj, fk, fl, fm, fn, fo, fp, fq, fr, fs, ft, fu, fv)(f)
+
+  def ^^^^^^^^^^^^^^^^^^^[A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W,X](fa: => F[A], fb: => F[B], fc: => F[C], fd: => F[D], fe: => F[E], fi: => F[I], fj: => F[J], fk: => F[K], fl: => F[L], fm: => F[M], fn: => F[N], fo: => F[O], fp: => F[P], fq: => F[Q], fr: => F[R], fs: => F[S], ft: => F[T], fu: => F[U], fv: => F[V], fw: => F[W])(
+                                                   f: (A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W) => X): F[X] =
+    F.apply20(fa, fb, fc, fd, fe, fi, fj, fk, fl, fm, fn, fo, fp, fq, fr, fs, ft, fu, fv, fw)(f)
+
+  def ^^^^^^^^^^^^^^^^^^^^[A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W,X,Y](fa: => F[A], fb: => F[B], fc: => F[C], fd: => F[D], fe: => F[E], fi: => F[I], fj: => F[J], fk: => F[K], fl: => F[L], fm: => F[M], fn: => F[N], fo: => F[O], fp: => F[P], fq: => F[Q], fr: => F[R], fs: => F[S], ft: => F[T], fu: => F[U], fv: => F[V], fw: => F[W], fx: => F[X])(
+                                                     f: (A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W,X) => Y): F[Y] =
+    F.apply21(fa, fb, fc, fd, fe, fi, fj, fk, fl, fm, fn, fo, fp, fq, fr, fs, ft, fu, fv, fw, fx)(f)
+
+  def ^^^^^^^^^^^^^^^^^^^^^[A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W,X,Y,Z](fa: => F[A], fb: => F[B], fc: => F[C], fd: => F[D], fe: => F[E], fi: => F[I], fj: => F[J], fk: => F[K], fl: => F[L], fm: => F[M], fn: => F[N], fo: => F[O], fp: => F[P], fq: => F[Q], fr: => F[R], fs: => F[S], ft: => F[T], fu: => F[U], fv: => F[V], fw: => F[W], fx: => F[X], fy: => F[Y])(
+                                                       f: (A,B,C,D,E,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W,X,Y) => Z): F[Z] =
+    F.apply22(fa, fb, fc, fd, fe, fi, fj, fk, fl, fm, fn, fo, fp, fq, fr, fs, ft, fu, fv, fw, fx, fy)(f)
+
```

Comment:
At some point we have to admit that this syntax if friggin ridiculous for n > 7 (ish, there's no real set point, but it's relatively obvious that this should not be a thing. The syntactic overhead is peak.

Reply:
The syntactic overhead is those peaks.